### PR TITLE
add retries for batch not suitable (WP-807)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/ApiWrapperInterface.php [hunk 1]
```diff
@@ -54,7 +62,6 @@ public function releaseLock(ConfigurationProfileEntity $profile, string $key): v
 
     public function createAuditLogRecord(
         ConfigurationProfileEntity $profile,
-        /** @noinspection PhpLanguageLevelInspection */
```
Comment (vsolovei-smartling) [inline inc/Smartling/ApiWrapperInterface.php:57]: Minimum version is php 8, this is redundant

File: inc/Smartling/ApiWrapper.php [hunk 2]
```diff
@@ -543,24 +544,32 @@ public function retrieveBatch(ConfigurationProfileEntity $profile, string $jobId
...
 
-    public function retrieveJobInfoForDailyBucketJob(ConfigurationProfileEntity $profile, bool $authorize): JobEntityWithBatchUid
+    private function getBaseNameForDailyBucketJob(string $suffix = ''): string
+    {
+        $date = date('m/d/Y');
+
+        return ApiWrapperInterface::DAILY_BUCKET_JOB_NAME_PREFIX . " $date$suffix";
+    }
+
+    public function findJobByBatchUid(ConfigurationProfileEntity $profile, string $batchUid): ?JobEntityWithStatus
```
Comment (dimitrystd) [inline inc/Smartling/ApiWrapper.php:554]: Jobs API doesn't know about batches. I don't understand how it can work.

**Review by dimitrystd — CHANGES_REQUESTED** [review]

File: inc/Smartling/ApiWrapper.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/ApiWrapper.php]: I've misread the parameters, you are correct. Fixed.

File: inc/Smartling/ApiWrapper.php [hunk 3]
```diff
@@ -543,24 +544,32 @@ public function retrieveBatch(ConfigurationProfileEntity $profile, string $jobId
...
     {
-        $getName = static function ($suffix = '') {
-            $date = date('m/d/Y');
-            $name = "Daily Bucket Job $date";
+        $parameters = new SearchJobsParameters();
+        $parameters->setFileUris([$fileUri]);
 
-            return $name . $suffix;
-        };
+        $result = $this->getJobsApi($profile)->searchJobs($parameters);
```
Comment (dimitrystd) [inline inc/Smartling/ApiWrapper.php:559]: We know that the same file can be uploaded many times. Please sort jobs by time Desc.

**Review by dimitrystd — CHANGES_REQUESTED** [review]

File: inc/Smartling/ApiWrapper.php (hunk 3 above)
Comment (vsolovei-smartling) [reply inc/Smartling/ApiWrapper.php]: Done

File: inc/Smartling/ApiWrapper.php [hunk 4]
```diff
@@ -543,24 +544,37 @@ public function retrieveBatch(ConfigurationProfileEntity $profile, string $jobId
...
 
-            return $name . $suffix;
-        };
+    public function findLastJobByFileUri(ConfigurationProfileEntity $profile, string $fileUri): ?JobEntityWithStatus
+    {
+        $parameters = new SearchJobsParameters();
+        $parameters->setFileUris([$fileUri]);
 
-        $jobName = $getName();
+        $result = array_reduce($this->getJobsApi($profile)->searchJobs($parameters), static function (?array $carry, array $item): array {
```
Comment (dimitrystd) [inline inc/Smartling/ApiWrapper.php:559]: OMG, how complex the `sort` + `first` implementations look in php.

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**